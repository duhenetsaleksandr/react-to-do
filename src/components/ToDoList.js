import React from 'react';
import '../styles/todo-list.scss';
import ToDoListItem from "./ToDoListItem";

export default function ToDoList(props) {
    return (
        <ul className="todos__list">
            {props.todos.map((todo, index) => {
                return <ToDoListItem
                    todo={todo}
                    index={index}
                    key={todo.id}
                    onEdit={props.onEdit}
                />
            })}
        </ul>
    );
}
