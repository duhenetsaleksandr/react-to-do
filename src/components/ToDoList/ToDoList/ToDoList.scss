@use 'src/styles/colors';

.todos__list {
    overflow: auto;
    max-height: 505px;
    overflow-y: scroll;

    margin-right: -10px;
    padding-right: 5px;

    &::-webkit-scrollbar-thumb {
        border-radius: 10px;
        background-color: colors.$secondary-color;
    }

    &::-webkit-scrollbar {
        width: 5px;
        background-color: transparent;
    }
}
