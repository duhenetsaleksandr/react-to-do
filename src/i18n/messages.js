import { LOCALES } from './locales'

export const messages = {
    [LOCALES.ENGLISH]: {
        copyright: '©Copyright. All rights deserved.',
        button_add: 'Add task',
        button_edit: 'Edit task',
        placeholder: 'Enter new todo',
        not_todos: 'Not found todos!',
    },
    [LOCALES.RUSSIAN]: {
        copyright: '©Копирайт. Все права защищены.',
        button_add: 'Добавить',
        button_edit: 'Редактировать',
        placeholder: 'Введите новую задачу',
        not_todos: 'Задачи не найдены!',
    },
};
