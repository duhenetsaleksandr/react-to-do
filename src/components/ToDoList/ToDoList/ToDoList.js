import React from 'react';
import './ToDoList.scss';
import ToDoListItem from 'components/ToDoList/ToDoListItem';
import NotTodos from 'components/ToDoList/NotTodos';

const ToDoList = ({ todos, onEdit, toggleTodo, deleteTodo}) => (
    <ul className="todos__list">
        {!todos.length ? <NotTodos/> : todos.map((todo) => (
            <ToDoListItem
                todo={todo}
                key={todo.id}
                toggleTodo={toggleTodo}
                deleteTodo={deleteTodo}
                onEdit={onEdit}
            />
        ))}
    </ul>
);

export default ToDoList;
