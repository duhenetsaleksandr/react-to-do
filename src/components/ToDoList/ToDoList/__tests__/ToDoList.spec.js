import React from 'react';
import ToDoList from '../ToDoList';
import { IntlProvider } from 'react-intl';
import { messages } from 'i18n/messages';
import { LOCALES } from 'i18n/locales';

const todos = [
    {
        id: 1,
        title: 'create app',
        completed: false
    },
    {
        id: 2,
        title: 'create app one more time',
        completed: true
    }
];

const props = {
    todos,
    onEdit: jest.fn(),
    toggleTodo: jest.fn(),
    deleteTodo: jest.fn(),
};

describe('should render ToDoList component', () => {
    describe('with default props', () => {
        let component;
        beforeEach(() => {
            component = mount(<ToDoList {...props} />);
        });

        it('should contain ToDoListItem component', () => {
            const toDoListItem = component.find('ToDoListItem');
            expect(toDoListItem).toHaveLength(todos.length);
        });

        it('should first ToDoListItem component contain all props', () => {
            const toDoListItem = component.find('ToDoListItem').at(0);
            const allProps = toDoListItem.getElement().props;
            expect(allProps.todo).toEqual(todos[0]);
            expect(allProps.onEdit).toEqual(props.onEdit);
            expect(allProps.toggleTodo).toEqual(props.toggleTodo);
            expect(allProps.deleteTodo).toEqual(props.deleteTodo);
        });

        it('should render ToDoList component snapshot with todos', () => {
            const component = shallow(<ToDoList {...props} />);
            expect(component).toMatchSnapshot();
        });
    });

    describe('with custom props', () => {
        it('should contain NotTodos component if todos array is empty', () => {
            const component = mount(<ToDoList {...props} todos={[]} />, {
                wrappingComponent: IntlProvider,
                wrappingComponentProps: {
                    messages: messages[LOCALES.ENGLISH],
                    locale: LOCALES.ENGLISH,
                    defaultLocale: LOCALES.ENGLISH,
                },
            });
            const notTodos = component.find('NotTodos');
            expect(notTodos).toHaveLength(1);
        });

        it('should render ToDoList component snapshot without todos', () => {
            const component = shallow(<ToDoList {...props} todos={[]} />);
            expect(component).toMatchSnapshot();
        });
    });
});
