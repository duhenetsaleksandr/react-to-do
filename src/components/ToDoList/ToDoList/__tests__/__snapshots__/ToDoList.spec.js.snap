// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ToDoList component with custom props should render ToDoList component snapshot without todos 1`] = `
<ul
  className="todos__list"
>
  <NotTodos />
</ul>
`;

exports[`should render ToDoList component with default props should render ToDoList component snapshot with todos 1`] = `
<ul
  className="todos__list"
>
  <ToDoListItem
    deleteTodo={[MockFunction]}
    key="1"
    onEdit={[MockFunction]}
    todo={
      Object {
        "completed": false,
        "id": 1,
        "title": "create app",
      }
    }
    toggleTodo={[MockFunction]}
  />
  <ToDoListItem
    deleteTodo={[MockFunction]}
    key="2"
    onEdit={[MockFunction]}
    todo={
      Object {
        "completed": true,
        "id": 2,
        "title": "create app one more time",
      }
    }
    toggleTodo={[MockFunction]}
  />
</ul>
`;
